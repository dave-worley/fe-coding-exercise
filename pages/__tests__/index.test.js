import autoJest from 'autotest/jest';
import autoStorybook from 'autotest/storybook';

import ComponentBase from '../index';
import mock from './data/mock.json';

export const Component = ComponentBase;

export const title = 'Page: Data';
export const exampleProps = {};

export const useCases = [
  {
    description: 'Default',
    props: {
      ...exampleProps,
      content: mock.data,
    },
  },
  {
    description: 'Not Found',
    props: {
      url: {
        query: {
          url: '/foo',
        },
      },
      error: { status: 404 },
    },
  },
  {
    description: '500 Error',
    props: {
      url: {
        query: {
          url: '/foo',
        },
      },
      error: { status: 500 },
    },
  },
];

autoJest(module.exports);
autoStorybook(module.exports);
