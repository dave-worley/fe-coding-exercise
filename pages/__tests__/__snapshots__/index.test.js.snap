// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Data: 500 Error 1`] = `
<div>
  It Broke
</div>
`;

exports[`Page: Data: Default 1`] = `
<React.Fragment>
  <Header />
  <TwoColumn
    data-article-body={true}
  >
    <Styled(div)>
      Here are things!
    </Styled(div)>
    <Styled(section)>
      And maybe other things?
    </Styled(section)>
  </TwoColumn>
</React.Fragment>
`;

exports[`Page: Data: Not Found 1`] = `
<div>
  It Broke
</div>
`;
